cocos/core/gfx/webgl2/webgl2-device.ts

```diff
@@ -140,53 +145,33 @@ export class WebGL2Device extends Device {
         const version: string = gl.getParameter(gl.VERSION);
 
         this._features.fill(false);
-        this._features[Feature.TEXTURE_FLOAT] = true;
-        this._features[Feature.TEXTURE_HALF_FLOAT] = true;
-        this._features[Feature.FORMAT_R11G11B10F] = true;
-        this._features[Feature.FORMAT_SRGB] = true;
-        this._features[Feature.FORMAT_RGB8] = true;
+
+        this.initFormatFeatures(exts);
+
         this._features[Feature.ELEMENT_INDEX_UINT] = true;
         this._features[Feature.INSTANCED_ARRAYS] = true;
         this._features[Feature.MULTIPLE_RENDER_TARGETS] = true;
         this._features[Feature.BLEND_MINMAX] = true;
 
-        if (exts.EXT_color_buffer_float) {
-            this._features[Feature.COLOR_FLOAT] = true;
-            this._features[Feature.COLOR_HALF_FLOAT] = true;
-        }
-
-        if (exts.OES_texture_float_linear) {
-            this._features[Feature.TEXTURE_FLOAT_LINEAR] = true;
-        }
-
-        if (exts.OES_texture_half_float_linear) {
-            this._features[Feature.TEXTURE_HALF_FLOAT_LINEAR] = true;
-        }
-
         let compressedFormat = '';
 
         if (exts.WEBGL_compressed_texture_etc1) {
-            this._features[Feature.FORMAT_ETC1] = true;
             compressedFormat += 'etc1 ';
```

Comment:
```suggestion
        if (this.getFormatFeatures(Format.ETC_RGB8)) {
            compressedFormat += 'etc1 ';
```

Reply:
Would be better to unify all these conditions to use the result from format feature interface instead of doing it all over again. This applies to the native engine as well.